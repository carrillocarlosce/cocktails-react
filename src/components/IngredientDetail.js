import React from "react";
import Tooltip from "@material-ui/core/Tooltip";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  tooltip: {
    cursor: "pointer"
  }
};

const IngredientDetail = function({ item, allIngredients, classes }) {
  if (item.special) return <span>{item.special}</span>;
  else {
    const { taste, abv } = allIngredients[item.ingredient] || {};
    const toolTipContent = [];
    if (abv > 0) {
      toolTipContent.push(abv + "% abv");
    } else {
      toolTipContent.push("Non-alcoholic");
    }
    if (taste) toolTipContent.push(taste);

    return (
      <span>
        {item.amount} {item.unit}{" "}
        <Tooltip
          className={classes.tooltip}
          title={toolTipContent.join(", ")}
          placement="top"
        >
          <strong>{item.label || item.ingredient}</strong>
        </Tooltip>
      </span>
    );
  }
};

const mapStateToProps = state => ({
  allIngredients: state.db.ingredients
});

export default withStyles(styles)(connect(mapStateToProps)(IngredientDetail));
