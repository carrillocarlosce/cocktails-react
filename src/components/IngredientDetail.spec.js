import React from "react";
import IngredientDetail from "./IngredientDetail";
import renderer from "react-test-renderer";

it("renders special ingredients correctly", () => {
  const item = {
    special: "1 cup of sugar"
  };
  const tree = renderer.create(<IngredientDetail item={item} />);
  expect(tree).toMatchSnapshot();
});

it("renders normal ingredients correctly", () => {
  const item = {
    name: "Vodka",
    amount: 1,
    unit: "cl"
  };
  const tree = renderer.create(<IngredientDetail item={item} />);
  expect(tree).toMatchSnapshot();
});

it("uses label instead of name if provided", () => {
  const item = {
    label: "Smirnoff",
    amount: 1,
    unit: "cl"
  };
  const tree = renderer.create(<IngredientDetail item={item} />);
  expect(tree).toMatchSnapshot();
});
