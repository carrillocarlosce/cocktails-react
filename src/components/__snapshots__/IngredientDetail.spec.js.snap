// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders normal ingredients correctly 1`] = `
<span>
  1
   
  cl
   
  <strong />
</span>
`;

exports[`renders special ingredients correctly 1`] = `
<span>
  1 cup of sugar
</span>
`;

exports[`uses label instead of name if provided 1`] = `
<span>
  1
   
  cl
   
  <strong>
    Smirnoff
  </strong>
</span>
`;
